src/main/java/org/candlepin/subscriptions/marketplace/MarketplaceSubscriptionCollector.java

```
@@ -20,32 +20,57 @@
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 import java.util.Collections;
 import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 /**
  * Class responsible for communicating with the Marketplace API and fetching the subscription ID.
  */
 @Component
 public class MarketplaceSubscriptionCollector {
-    @SuppressWarnings("java:S1068") // Unused field; Remove after implementing
-    private final MarketplaceApi marketplaceApi;
+
+    private static final String IBMMARKETPLACE = "ibmmarketplace";
+    private static final Logger log = LoggerFactory.getLogger(MarketplaceSubscriptionCollector.class);
+    private final SubscriptionService subscriptionService;
 
     @Autowired
-    public MarketplaceSubscriptionCollector(MarketplaceApi marketplaceApi,
-        MarketplaceProperties properties) {
-        this.marketplaceApi = marketplaceApi;
+    public MarketplaceSubscriptionCollector(SubscriptionService subscriptionService) {
+        this.subscriptionService = subscriptionService;
+    }
+
+    /**
+     * Given an account number, query the IT Subscription Service and return subscriptions that have an ibm
+     * marketplace external reference as part of its payload
+     * @param accountNumber - account number aka oracle account number aka ebs account number
+     * @return subscriptions
+     */
+    public List<Subscription> requestSubscriptions(String accountNumber) {
+        try {
+            var subscriptions = subscriptionService.getSubscriptionsByAccountNumber(accountNumber);
+            return filterNonApplicableSubscriptions(subscriptions);
+        }
+        catch (ApiException e) {
+            log.error("Exception occurred when making GET request to subscription service via account " +
+                "number", e);
+            return Collections.emptyList();
```

I'd rather throw an exception here.